rclcpp/include/rclcpp/qos.hpp

```diff
@@ -90,18 +90,36 @@ struct RCLCPP_PUBLIC KeepLast : public rclcpp::QoSInitialization
 };
 
 /// Encapsulation of Quality of Service settings.
+/**
+ * Quality of Service settings contains parameters for the underlying middleware, typically DDS,
+ * to tune communication between nodes. See
+ * <a href="https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings/">
+ About Quality of Service Settings</a> for an explanation.
+ */
```

Comment:
```suggestion
/**
 * Quality of Service settings control the behavior of publishers, subscriptions,
 * and other entities, and includes things like how data is sent or resent,
 * how data is buffered on the publishing and subscribing side, and other things.
 * See:
 *   <a href="https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings">
 *     https://index.ros.org/doc/ros2/Concepts/About-Quality-of-Service-Settings
 *   </a>
 */
```

We should avoid mentioning DDS or any underlying middleware specifics at this level of documentation.

Also, I prefer if we don't hind url's behind anchors. The url is human readable and more transparent if it is exposed directly (you know where you're going without mouse hovering).